pkgs/checks/test/extensions/async_test.dart

```diff
@@ -6,39 +6,71 @@ import 'dart:async';
 
 import 'package:async/async.dart';
 import 'package:checks/checks.dart';
+import 'package:checks/context.dart';
 import 'package:test/scaffolding.dart';
 import 'package:test_api/hooks.dart';
 
 import '../test_shared.dart';
 
 void main() {
   group('FutureChecks', () {
-    test('completes', () async {
-      (await checkThat(_futureSuccess()).completes()).equals(42);
-
-      await checkThat(_futureFail()).isRejectedByAsync(
-        it()..completes().that(it()..equals(1)),
-        actual: ['A future that completes as an error'],
-        which: ['Threw <UnimplementedError>'],
-      );
+    group('completes', () {
+      test('succeeds for a future that completes to a value', () async {
+        await checkThat(_futureSuccess()).completes().that(it()..equals(42));
```

Comment:
meta comment: the `.that` here reads a bit weird to me here and below, feels more like a `with`, `where`, or `which`?

Reply:
I have a doc which had a section considering this. The same suggestion was made an on earlier PR. I think that's probably a good indication that it's worth chagning.